src/rgw/driver/rados/rgw_sal_rados.h

```diff
@@ -157,6 +160,21 @@ class RadosStore : public StoreDriver {
 	optional_yield y, const DoutPrefixProvider *dpp) override;
     int remove_topics(const std::string& tenant, RGWObjVersionTracker* objv_tracker,
         optional_yield y, const DoutPrefixProvider *dpp) override;
+    int read_topic(const std::string& topic_name,
```

Comment:
if these are the v2 versions of the API, maybe it is better to cal them `read_topic2()` etc.
the singular vs. plural may be confusing, because, in theory, we can have `read_topic()` etc. of v1 topics as well

Reply:
The reason i did not change the name's of the pubsub API's (read,write,get,delete), so that we do not have the change the interface... all the callers can still call the same API's, pubsub will handle the v1 and v2.